site/gatsby-site/migrations/2022.03.04T15.44.49.incidents-normalization.js

```diff
@@ -0,0 +1,107 @@
+const lodash = require('lodash');
+
+/**
+ *
+ * @param {{context: {client: import('mongodb').MongoClient}}} context
+ */
+
+exports.up = async ({ context: { client } }) => {
+  await client.connect();
+
+  const incidentsCollection = client.db('aiidprod').collection('incidents');
+
+  const reportsCollection = client.db('aiidprod').collection('reports');
+
+  const incidents = await incidentsCollection.find({}).toArray();
+
+  const incidentReportGroups = lodash.groupBy(incidents, 'incident_id');
+
+  for (const key in incidentReportGroups) {
+    const group = incidentReportGroups[key];
+
+    const { incident_date } = group[0];
+
+    const incidentId = parseInt(key);
+
+    const reportIds = group.map((r) => r.report_number);
+
+    const newIncident = { incident_id: incidentId, date: incident_date, reports: reportIds };
+
+    console.log('delete old incidents with id:', { incident_id: incidentId });
+    await incidentsCollection.deleteMany({ incident_id: incidentId });
+
+    console.log(`new incident`, newIncident);
+    await incidentsCollection.insertOne(newIncident);
+
+    const reports = [];
+
+    for (const report of group) {
+      const {
+        _id,
+        incident_id,
+        cloudinary_id,
+        authors,
+        date_downloaded,
+        date_modified,
+        date_published,
+        date_submitted,
+        description,
+        epoch_date_downloaded,
+        epoch_date_modified,
+        epoch_date_published,
+        epoch_date_submitted,
+        epoch_incident_date,
```

Comment:
This should be on the incident collection, not the reports collection.

Reply:
Yes, same as my comment above, I was thinking to update it in another PR, too many changes at once otherwise.